src/mavlink-router/endpoint.h

```diff
@@ -131,6 +134,9 @@ class Endpoint : public Pollable {
     const bool _crc_check_enabled;
     uint32_t _incomplete_msgs = 0;
     std::vector<uint16_t> _sys_comp_ids;
+
+private:
+    std::set<uint32_t> message_filter;
```

Comment:
I'm not a big fan of std::set. I'd rather use std::vector here. When we are filtering we will probably care for very few messages, so std::vector should even be faster for lookups.

Reply:
Replaced std::set with std::vector.